examples/threadExample/src/main/java/io/realm/examples/threads/AsyncQueryFragment.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.examples.threads;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.Collections;
+import java.util.List;
+
+import io.realm.Realm;
+import io.realm.RealmChangeListener;
+import io.realm.RealmResults;
+import io.realm.examples.threads.model.Dot;
+
+/**
+ * This fragment demonstrates how you can perform asynchronous queries with Realm
+ */
+public class AsyncQueryFragment extends Fragment implements View.OnClickListener, RealmChangeListener {
+    private Realm realm;
+    private DotAdapter dotAdapter;
+    private RealmResults<Dot> allSortedDots;
+    private Realm.Request asyncTransaction;
+
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_async_query, container, false);
+        rootView.findViewById(R.id.translate_button).setOnClickListener(this);
+
+        ListView listView = (ListView) rootView.findViewById(android.R.id.list);
+        dotAdapter = new DotAdapter(getActivity());
+        listView.setAdapter(dotAdapter);
+        return rootView;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // Create Realm instance for the UI thread
+        realm = Realm.getDefaultInstance();
+        allSortedDots = realm.where(Dot.class)
+                .between("x", 25, 75)
+                .between("y", 0, 50)
+                .findAllSortedAsync(
+                         "x", RealmResults.SORT_ORDER_ASCENDING,
+                         "y", RealmResults.SORT_ORDER_DESCENDING
+                 );
+        dotAdapter.updateList(allSortedDots);
+        allSortedDots.addChangeListener(this);
```

Comment:
I am a bit nervous about this. after `findAllSortedAsync` return and before `allSortedDots.addChangeListener(this);` called, there is a chance that query is finished. In that case, listener won't be called anymore, right?